packages/client-engine-runtime/src/interpreter/renderQuery.ts

```diff
@@ -15,16 +27,62 @@ export function renderQuery({ query, params }: QueryPlanDbQuery, scope: ScopeBin
       throw new Error(`Missing value for query variable ${param.prisma__value.name}`)
     }
 
-    return value
+    return value as PrismaValue
   })
+}
+
+function renderTemplateSql(
+  fragments: Fragment[],
+  placeholderFormat: PlaceholderFormat,
+  params: PrismaValue[],
+): SqlQuery {
+  let paramIndex = 0
+  let placeholderNumber = 1
+  const flattenedParams: PrismaValue[] = []
+  const sql = fragments
+    .map((fragment) => {
+      switch (fragment.type) {
+        case 'parameter':
+          flattenedParams.push(params[paramIndex++])
+          return formatPlaceholder(placeholderFormat, placeholderNumber++)
+
+        case 'stringChunk':
+          return fragment.value
+
+        case 'parameterTuple': {
+          const paramValue = params[paramIndex++]
+          const paramArray = Array.isArray(paramValue) ? paramValue : [paramValue]
+          const placeholders =
+            paramArray.length == 0
+              ? 'NULL'
+              : paramArray
+                  .map((value) => {
+                    flattenedParams.push(value)
+                    return formatPlaceholder(placeholderFormat, placeholderNumber++)
+                  })
+                  .join(', ')
```

Comment:
Actually no, sorry, I was wrong. I thought we were formatting it like this but it's not the case, e.g.:
https://github.com/prisma/prisma/blob/0fa294aa0c67759375aeb92de78c6dc6b3135338/packages/client/tests/functional/0-legacy-ports/batch-find-unique/tests.ts#L87-L95

Let's revert this change:
```suggestion
                  .join(',')
```

Reply:
No problem. Removed the space.